nixos/modules/services/web-apps/archtika.nix

```diff
@@ -0,0 +1,312 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+with lib;
+
+let
+  cfg = config.services.archtika;
+  baseHardenedSystemdOptions = {
+    CapabilityBoundingSet = "";
+    LockPersonality = true;
+    NoNewPrivileges = true;
+    PrivateDevices = true;
+    PrivateTmp = true;
+    ProtectClock = true;
+    ProtectControlGroups = true;
+    ProtectHome = true;
+    ProtectHostname = true;
+    ProtectKernelLogs = true;
+    ProtectKernelModules = true;
+    ProtectKernelTunables = true;
+    ProtectSystem = "strict";
+    RemoveIPC = true;
+    RestrictNamespaces = true;
+    RestrictRealtime = true;
+    RestrictSUIDSGID = true;
+    SystemCallArchitectures = "native";
+    SystemCallFilter = [
+      "@system-service"
+      "~@privileged"
+      "~@resources"
+    ];
+
+    ReadWritePaths = [ "/var/www/archtika-websites" ];
+  };
+in
+{
+  options.services.archtika = {
+    enable = mkEnableOption "archtika service";
+
+    package = mkPackageOption pkgs "archtika" { };
+
+    user = mkOption {
+      type = types.str;
+      default = "archtika";
+      description = "User account under which archtika runs.";
+    };
+
+    group = mkOption {
+      type = types.str;
+      default = "archtika";
+      description = "Group under which archtika runs.";
+    };
```

Comment:
Might be a stupid question (I don't know how Archtika works), but could this be replaced by [those fancy dynamic users systemd makes](https://0pointer.net/blog/dynamic-users-with-systemd.html)? Do we need to explicitly create an user here?

Reply:
archtika needs shared access between several services. For example, I wanted an easy way for the `postgres` and `nginx` users to also have read and write access to the `/var/www/archtika-websites` directory, which is easily done with a static group:

```nix
users.groups.${cfg.group} = {
  members = [...
    "nginx
    "postgres
  ];
};
```

As for the user, an explicit one could probably be omitted, but I think it does no harm to leave it there. Also, I like the idea that the `/var/www/archtika-websites` directory and all its contents are owned by both the archtika user and the group.

Reply:
That's alright then! Leave it as is.